common/src/execution_mode.rs

```diff
@@ -0,0 +1,18 @@
+#[derive(Clone)]
+pub enum ExecutionMode {
+    Rpc(String),
+    VerifiableApi(String),
+}
+
+impl ExecutionMode {
+    pub fn from_urls(rpc: Option<String>, verifiable_api: Option<String>) -> Self {
+        match (rpc, verifiable_api) {
+            (None, Some(verifiable_api)) => Self::VerifiableApi(verifiable_api),
+            (Some(rpc), None) => Self::Rpc(rpc),
+            (Some(_), Some(_)) => {
```

Comment:
Maybe if both are available default to the verifiable api?

Reply:
Can do. Maybe could make sense to log a message at warn/info level informing user about it too.

Reply:
I don't think there is a need